include/nlohmann/detail/json_pointer.hpp

```diff
@@ -344,10 +348,10 @@ class json_pointer
         }
 
         std::size_t processed_chars = 0;
-        int res = 0;
+        size_type res = 0;
         JSON_TRY
         {
-            res = std::stoi(s, &processed_chars);
+            res = std::stoull(s, &processed_chars);
```

Comment:
Don't we need to cast the result of `std::stoull` to `size_type` here?

Reply:
@nlohmann 

Yes and no. Yes casting will prevent a warning when `size_type` and `unsigned long long` don't coincide. But on the other hand if they don't coincide (32bit platform) size_type should be smaller than `unsigned long long` thus breaking it altogether.

Maybe I should do more something like the following?

```cpp

unsigned long long res = std::stoull(s, &processed_chars);

if(res >= static_cast<unsigned long long>(std::numeric_limits<size_type>::max()))
{
throw ...
}

return static_cast<size_type>(res);
````

Reply:
Yes, this looks good. I think throwing `json::out_of_range` would make sense. We don't have an exception id for this yet, so I would propose using 410, e.g. like

```cpp
JSON_THROW(out_of_range::create(410, "array index " + s + " exceeds size_type");
```

Can you make that change?

Reply:
@nlohmann Done. Writing a test for it was a bit tricky though.

Reply:
@nlohmann Done.